src/application/gitcoin/hooks/use-recipients.ts

```diff
@@ -0,0 +1,72 @@
+import { useMemo } from 'react';
+
+import { useCommandQuery } from 'shared/messaging';
+import { ScrapingResult } from 'shared/scraping';
+import { isHandleNode } from 'host/twitter'; // TODO: abstract this so it's not specific to Twitter, probably we need to accept it via Properties
+
+import { GetApplicationsCommand } from '../commands';
+import { selectTwitterApplications } from '../utils';
+import { Recipient } from '../types';
+
+interface Properties {
+  users: ScrapingResult[];
+  handle?: string;
+  enabled: boolean;
+}
+
+export const useRecipients = ({ users, handle, enabled }: Properties) => {
+  const getApplicationsQuery = useCommandQuery({
+    command: new GetApplicationsCommand({}),
+    select: selectTwitterApplications,
+    placeholderData: (previousData) => {
+      return previousData;
+    },
+    enabled,
+  });
+
+  const recipients: Recipient[] = useMemo(() => {
+    if (!getApplicationsQuery.data) {
+      return [];
+    }
+
+    return users
+      .map((user) => {
+        const application = getApplicationsQuery.data.find((application) => {
+          return (
+            application.project.metadata.projectTwitter?.toLowerCase() ===
+            user.value.toLowerCase()
+          );
+        });
+
+        if (!application) {
+          return;
+        }
+
+        const { value: username, node, top } = user;
+
+        const isHandleUser =
+          handle === username && isHandleNode(node as HTMLElement);
+
+        const nodeToInject: HTMLElement | null = isHandleUser
+          ? node.querySelector('div > div > div > span') ??
```

Comment:
What do you think about moving the paths to shared constants?

Reply:
good idea, I moved it to host/twitter/utils, it would be hard to find some meaningful name for constants for this :D